### PR TITLE
Exclude "scripts" from setup package

### LINKED ISSUE #4005: scripts/ is getting installed in site-packages
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04
- **Modin version** (`modin.__version__`): 0.12.0
- **Python version**:  3.8
- **Code we can use to reproduce**:

I believe the scripts/ folder is getting installed in site-packages with modin. I used ```pip install modin[dask]```. We ran into an import issue since we have a scripts module (conflict between using ours vs the site-packages installed.) but when we install modin we found that your scripts module is getting installed in the site-packages. I am not sure if this is a mistake or intentional.

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?
Exclude `scripts` and its subdirectories from setup package.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4005
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     version=versioneer.get_version(),
     cmdclass=versioneer.get_cmdclass(),
     description="Modin: Make your pandas code run faster by changing one line of code.",
-    packages=find_packages(),
+    packages=find_packages(exclude=["scripts", "scripts.*"]),
```
Comment (YarShev) [inline setup.py:19]: Even though our "scripts" folder is used for testing only, should we move it under "modin" directory so "scripts" is accessible via "modin.scripts"?

Comment (vnlitvinov) [reply setup.py:19]: That is another option, I agree. But I don't see much value in that.

Comment (YarShev) [reply setup.py:19]: We may put there something useful in future so that might be accessible when installing Modin package. Does that make sense?

Comment (vnlitvinov) [reply setup.py:19]: In this case we'll move `scripts` when needed, and this exclude won't prevent it from inclusion 😉

**Review by devin-petersohn — APPROVED** [review]
LGTM, This is a good initial solution.

File: setup.py (hunk 1 above)
Comment (devin-petersohn) [reply setup.py:19]: Yes, this is a one line change rather than a refactor, so I am in favor of this change first

Comment (YarShev) [reply setup.py:19]: Okay, makes sense.

**Review by YarShev — APPROVED** [review]

**Pull request merged**